gqlalchemy/graph_algorithms/integrated_algorithms.py

```diff
@@ -201,3 +202,54 @@ def __str__(self) -> str:
             algo_str = f"{algo_str} {filter_lambda}"
 
         return algo_str
+
+
+class AllShortestPath(IntegratedAlgorithm):
+    """Build a Djikstra ALL shortest paths call for a Cypher query.
+    The all weighted shortest paths algorithm can"""
+
+    """Build a Djikstra shortest path call for a Cypher query
+    The weighted shortest path algorithm can be called in Memgraph with Cypher
+    queries such as:
+    " MATCH (a {id: 723})-[r *ALLSHORTEST 10 (r, n | r.weight) total_weight
+            (r, n | r.x > 12 AND r.y < 3)]-(b {id: 882}) RETURN * "
+    It is called inside the relationship clause, "*ALLSHORTEST" naming the
+    algorithm, "10" specifying search depth bounds, and "(r, n | <expression>)"
+    is a filter lambda, used to filter which relationships and nodes to use.
+    """
+
+    def __init__(
+        self,
+        upper_bound: int = None,
+        condition: str = None,
+        total_weight_var: str = DEFAULT_TOTAL_WEIGHT,
+        weight_property: str = DEFAULT_WEIGHT_PROPERTY,
+    ) -> None:
+        """
+        Args:
+            upper_bound: Upper bound for path depth. Defaults to None.
+            condition: Filter through nodes and relationships that pass this
+            condition. Defaults to None.
+            total_weight_var: Variable defined as the sum of all weights on
+            path being returned. Defaults to "total_weight".
+            weight_property: property being used as weight. Defaults to
+            "r.weight".
+        """
+        super().__init__()
+        self.weight_property = f"r.{weight_property}" if "." not in weight_property else weight_property
+        self.total_weight_var = total_weight_var
+        self.condition = condition
+        self.upper_bound = str(upper_bound) if upper_bound is not None else ""
```

Comment:
```suggestion
        self.upper_bound = "" if upper_bound is None else str(upper_bound)
```

Again the same. Correct my code if I am wrong, writing in comments.